### DISCUSSION TITLE
How to add a parameters to a file

### DISCUSSION THREAD

**Discussion by Yrow0** [artgris/FileManagerBundle#122] [Q&A]
HI, at first sorry i am french and my english is pretty so forgive me please.

I am currently using your bundle and I want to know if i can add a new parameter to the files that will be the user who upload the file or if this is currently stocked. So i want to know how can i add a parameter in your bundle or How to access who is the user who uploads the file. 

For more details i am using the native symfony way to manage user.

**Comment by artgris** [comment]
Hi,

You can access the user either through events, for example, using POST_UPDATE :

```
<?php

namespace App\Event;

use Artgris\Bundle\FileManagerBundle\Event\FileManagerEvents;
use Symfony\Component\EventDispatcher\EventSubscriberInterface;
use Symfony\Component\EventDispatcher\GenericEvent;
use Symfony\Bundle\SecurityBundle\Security;

class FileManagerSubscriber implements EventSubscriberInterface
{

    public function __construct(private Security $security)
    {
    }

    public static function getSubscribedEvents(): array
    {
        return [
            FileManagerEvents::POST_UPDATE => 'onPostUpdate',
        ];
    }

    public function onPostUpdate(GenericEvent $event)
    {
        $user = $this->security->getUser();
        dd($user);

    }

}
```
or through [the service configurator](https://github.com/artgris/FileManagerBundle/blob/master/Resources/doc/book/2-service-configuration.md) to modify the file manager configuration based on the user.

**Reply by Yrow0** [reply]
Hi, 

and after getting the user, how can i show or not the delete and edit button. Sry i am currently a beginner.

Thank you

**Comment by artgris** [comment]
In this case, I think it would be necessary to override the templates to show or hide buttons (relative to the current user) and block edit and delete actions using the RENAME_FILE and PRE_DELETE_FILE events.
